### ISSUE TITLE
Change handling of Vagrantfiles to support Jinja templating

### ISSUE THREAD

**Issue by agperson** [huit/nepho#169]
Previously a processed Vagrantfile was dropped into CWD and the box launched from there, which could lead to confusion.  For the moment, we are just using the Vagrantfile inside the cloudlet and launching from that location (.nepho/cloudlets/&lt;cloudlet&gt;/resources/&lt;blueprint&gt;).  We should generate a Vagrantfile with Jinja and place it into a new "working" location along with the payload, named appropriately, and launch the VM from there.  This should also allow us to support launching multiple versions of the same blueprint using the Vagrant provider.

**Comment by robparrott** [comment]
Copy instead of move, right?

On November 20, 2013 at 6:05:07 PM, Daniel Silverman (notifications@github.com) wrote:

Previously a Vagrantfile was dropped into CWD and the box launched from there. For the moment, we are just using the Vagrantfile inside the cloudlet and launching from there. We should move the template-processed Vagrantfile into a working directory along with the payload, then run it from there.

**Comment by agperson** [comment]
Updated description to be more clear.

**Comment by hakamadare** [comment]
btw, look at the `root` parameter that can be passed to `Vagrant.__init__()` in python-vagrant; i think that's a way to specify the working directory other than CWD.

https://github.com/todddeluca/python-vagrant/blob/master/vagrant.py

``` python
def __init__(self, root=None):
    '''
    root: a directory containing a file named Vagrantfile.  Defaults to
    os.getcwd(). This is the directory and Vagrantfile that the Vagrant
    instance will operate on.
    '''
    self.root = os.path.abspath(root) if root is not None else os.getcwd()
    self._cached_conf = {}
```

**Comment by hakamadare** [comment]
yeah, either the root param doesn't do what i think, or i'm not calling it correctly.  also it looks like the VagrantProvider parameter at least (and possibly BoxName and BoxUrl) are not being properly handled (i.e. setting them has no effect on nepho's behavior.

this looks awfully like work, slipping it to 1.1.0.

**Comment by agperson** [comment]
This was added in #221 and released in 1.2.0
